### PR TITLE
Feat: Add workspace level invite flow, enhance UI, and refactor workspace explorer [#1086]

### PR DESCRIPTION
**Pull request by itsmdasifraza**
## Description

This pull request introduces several key updates:

1. **Workspace Level Invite Flow**
    - Added the ability to invite users to a workspace with required parameters: email and role (editor or viewer).
    - Role options:
        - `editor`: Can edit the workspace.
        - `viewer`: Can view the workspace.

2. **UI Enhancements**
    - Improved the overall UI for a better user experience.
    - Updated styles and layouts for consistency and usability.

3. **Code Refactor**
    - Refactored the `workspace explorer` module into component-wise breakdown for better maintainability and readability.
    - Improved code structure and organization.

4. **Library**
    - Introduced `MultiSelect` component in a app library 

Fixes issue #1086 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

**Comment by pratikshinde590** [comment]
Please attach the issue link

**Comment by pratikshinde590** [comment]
Please attach the issue link

**Review by gc-codes — APPROVED** [review]

File: src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.ViewModel.ts [hunk 1]
```diff
@@ -157,4 +172,35 @@ export default class WorkspaceExplorerViewModel {
...
+   * Invites users to a workspace.
+   * @param currentWorkspaceDetails The details of the current workspace.
+   * @param data The payload containing users and their roles.
+   * @param emailstoBeSentArr The array of email addresses to send invitations to.
+   * @returns A promise resolving to the response from the invitation operation.
+   */
+  public inviteUserToWorkspace = async (
+    currentWorkspaceDetails,
+    data,
+    emailstoBeSentArr,
```
Comment (Astitva877) [inline src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.ViewModel.ts:186]: is it possible to give types here ?

File: src/packages/@library/forms/multi-select/MultiSelect.svelte [hunk 2]
```diff
@@ -0,0 +1,285 @@
+<script lang="ts">
+  import dropdown from "$lib/assets/dropdown.svg";
+  import { onDestroy, onMount } from "svelte";
+  import { slide } from "svelte/transition";
+  import closeIcon from "$lib/assets/close.svg";
+
+  interface Data {
+    name: string;
+    id: string;
+    checked: boolean;
```
Comment (Astitva877) [inline src/packages/@library/forms/multi-select/MultiSelect.svelte:10]: Shift these to a separate file.

File: src/packages/@library/forms/multi-select/MultiSelect.svelte [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+   */
+  const toggleDropdown = () => {
+    isOpen = !isOpen;
+  };
+
+  /**
+   * Handles clicks outside the dropdown to close it.
+   * @param  event - The mouse event.
+   */
+  function handleDropdownClick(event: MouseEvent) {
```
Comment (Astitva877) [inline src/packages/@library/forms/multi-select/MultiSelect.svelte:52]: make it a arrow function, which we are also following everywhere

File: src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+  import closeIconWhite from "$lib/assets/close-icon-white.svg";
+  import { createDynamicComponents } from "$lib/utils/helpers/common.helper";
+  import { onDestroy, onMount } from "svelte";
+  import { slide } from "svelte/transition";
+
+  /**
+   * Interface representing a user in the list.
+   */
+  interface List {
+    email: string;
```
Comment (Astitva877) [inline src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte:12]: shift these from here also

File: src/packages/@library/forms/multi-select/MultiSelect.svelte [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+  }
+  .highlight:hover {
+    background-color: #232424;
+  }
+  .dropdown-btn {
+    border: 1px solid #313233;
+    cursor: pointer;
+  }
+  .dropdown-btn-active {
+    border: 1px solid var(--send-button);
```
Comment (Astitva877) [inline src/packages/@library/forms/multi-select/MultiSelect.svelte:276]: use --border here

File: src/packages/@library/forms/multi-select/MultiSelect.svelte [hunk 6]
```diff
@@ -0,0 +1,285 @@
...
+  }
+  .isError {
+    border: 1px solid var(--error--color) !important;
```
Comment (Astitva877) [inline src/packages/@library/forms/multi-select/MultiSelect.svelte:279]: use --border here

**Review by Astitva877 — CHANGES_REQUESTED** [review]

File: src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.svelte [hunk 7]
```diff
@@ -1,38 +1,25 @@
...
-  let showWorkspaceSettings = false;
-  let workspaceName = tab.name;
-  let collectionLength = 0;
-  let workspaceDescription = tab.description;
-  let workspaceID = tab._data.path.workspaceId;
+  const _viewModel = new WorkspaceExplorerViewModel();
+  const activeWorkspace: Observable<WorkspaceDocument> =
+    _viewModel.activeWorkspace;
+
+  let isWorkspaceInviteModalOpen: boolean = false;
```
Comment (Astitva877) [inline src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.svelte:18]: i don't think so we need to give here type specifically as we have already set a default value.

File: src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+    {/if}
+  </div>
+</div>
+
+<style>
+  .email-container {
+    background-color: var(--bg-tertiary-300);
+    border: 1px solid;
+    padding: 3px 8px 3px 8px;
+    border: 1px solid var(--border-color);
```
Comment (Astitva877) [inline src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte:233]: use standard format for border color like --border-primary-100

File: src/packages/@workspaces/features/invite-to-workspace/layout/InviteToWorkspace.svelte [hunk 9]
```diff
@@ -0,0 +1,203 @@
...
+      onClick={() => {
+        handleInvite();
+      }}
+    />
+  </div>
+</div>
+
+<style>
+  .asterik {
+    color: var(--dangerColor);
```
Comment (Astitva877) [inline src/packages/@workspaces/features/invite-to-workspace/layout/InviteToWorkspace.svelte:196]: use standard format

File: src/packages/@workspaces/features/invite-to-workspace/layout/InviteToWorkspace.svelte [hunk 10]
```diff
@@ -0,0 +1,203 @@
...
+    margin-left: 4px;
+  }
+  .error-text {
+    margin-top: 4px;
+    color: var(--error--color);
```
Comment (Astitva877) [inline src/packages/@workspaces/features/invite-to-workspace/layout/InviteToWorkspace.svelte:201]: here also

File: src/packages/@library/forms/multi-select/MultiSelect.svelte (hunk 2 above)
Comment (itsmdasifraza) [reply src/packages/@library/forms/multi-select/MultiSelect.svelte:10]: resolved

File: src/packages/@library/forms/multi-select/MultiSelect.svelte (hunk 3 above)
Comment (itsmdasifraza) [reply src/packages/@library/forms/multi-select/MultiSelect.svelte:52]: resolved

File: src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte (hunk 4 above)
Comment (itsmdasifraza) [reply src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte]: resolved

File: src/packages/@library/forms/multi-select/MultiSelect.svelte (hunk 5 above)
Comment (itsmdasifraza) [reply src/packages/@library/forms/multi-select/MultiSelect.svelte]: resolved

File: src/packages/@library/forms/multi-select/MultiSelect.svelte (hunk 6 above)
Comment (itsmdasifraza) [reply src/packages/@library/forms/multi-select/MultiSelect.svelte]: resolved

File: src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.svelte (hunk 7 above)
Comment (itsmdasifraza) [reply src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.svelte]: resolved

File: src/packages/@workspaces/features/invite-to-workspace/layout/InviteToWorkspace.svelte (hunk 9 above)
Comment (itsmdasifraza) [reply src/packages/@workspaces/features/invite-to-workspace/layout/InviteToWorkspace.svelte]: resolved

File: src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte (hunk 8 above)
Comment (itsmdasifraza) [reply src/packages/@workspaces/features/invite-to-workspace/components/invite-user-pickers/InviteUserPicker.svelte]: resolved

File: src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.ViewModel.ts (hunk 1 above)
Comment (itsmdasifraza) [reply src/packages/@app/pages/WorkspaceExplorerPage/WorkspaceExplorerPage.ViewModel.ts]: resolved

**Review by Astitva877 — APPROVED** [review]

**Pull request merged**